radicle-cli/src/git.rs

```diff
@@ -175,24 +175,48 @@ pub fn is_signing_configured(repo: &Path) -> Result<bool, anyhow::Error> {
 }
 
 /// Return the list of radicle remotes for the given repository.
-pub fn remotes(repo: &git2::Repository) -> anyhow::Result<Vec<(String, NodeId)>> {
-    let mut remotes = Vec::new();
-
-    for name in repo.remotes().iter().flatten().flatten() {
-        let remote = repo.find_remote(name)?;
-        for refspec in remote.refspecs() {
-            if refspec.direction() != git2::Direction::Fetch {
-                continue;
+pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<git2::Remote>> {
+    let remotes: Vec<_> = repo
+        .remotes()?
+        .iter()
+        .filter_map(|name| {
+            let Some(name) = name else {
+                return None
+            };
+            let Some(remote) = repo.find_remote(name).ok() else {
+                return None
+            };
+
+            let Some(url) = remote.url() else {
+                return None
+            };
+
+            if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
+                Some(remote)
+            } else {
+                None
             }
-            if let Some((peer, _)) = refspec.src().and_then(self::parse_remote) {
-                remotes.push((name.to_owned(), peer));
-            }
-        }
-    }
-
+        })
+        .collect();
     Ok(remotes)
 }
 
+/// Return true is the remote is stored inside the  the list of radicle remotes.
+pub fn rad_has_remote(repo: &git2::Repository, alias: &str) -> anyhow::Result<bool> {
+    let remotes = repo.remotes()?;
+    let found = remotes.iter().any(|name| {
+        let Some(name) = name else {
+                return false;
+            };
+
+        if name == alias {
+            return true;
+        }
+        false
```

Comment:
What about just `name ==alias` in place of the if statement?

Reply:
yeah much better thanks :)